src/oneD/StFlow.cpp

```diff
@@ -20,18 +20,19 @@ StFlow::StFlow(ThermoPhase* ph, size_t nsp, size_t points) :
     Domain1D(nsp+c_offset_Y, points),
     m_nsp(nsp)
 {
-    if (ph->type() == "ideal-gas") {
-        m_thermo = static_cast<IdealGasPhase*>(ph);
-    } else {
-        throw CanteraError("StFlow::StFlow",
-                           "Unsupported phase type: need 'IdealGasPhase'");
-    }
     m_type = cFlowType;
     m_points = points;
 
     if (ph == 0) {
         return; // used to create a dummy object
     }
+    if (ph->type() == "ideal-gas" || ph->type() == "Redlich-Kwong" || 
+        ph->type() == "Peng-Robinson") {
+        m_thermo = static_cast<ThermoPhase*>(ph);
+    } else {
+        throw CanteraError("StFlow::StFlow", "Unsupported phase type:", 
+            "need 'IdealGasPhase' or 'Redlich-Kwong'or 'Peng-Robinson'");
+    }
```

Comment:
The phase check is not necessary. Without it, trying to instantiate `StFlow` based on the solid phase defined in `diamond.yaml` will fail because there is no valid transport model, which seems fine to me. You can just modify the test to match `transport model` instead.

Reply:
I have updated the test.